src/plugins/windowmon/windowmon.cpp

```diff
@@ -0,0 +1,344 @@
+
+    switch (c->format)
+    {
+        case OUTPUT_CSV:
+            printf("windowmon," FORMAT_TIMEVAL ",%" PRIu32 ",0x%" PRIx64 ",\"%s\",%" PRIi64 ",%s,%s\n",
+                   UNPACK_TIMEVAL(info->timestamp), info->vcpu, info->regs->cr3, info->proc_data.name,
+                   info->proc_data.userid, window_class, window_name);
+            break;
+        case OUTPUT_KV:
+            printf("windowmon Time=" FORMAT_TIMEVAL ",PID=%d,PPID=%d,ProcessName=\"%s\",Method=%s,Class=%s,Name=%s\n",
+                   UNPACK_TIMEVAL(info->timestamp), info->proc_data.pid, info->proc_data.ppid, info->proc_data.name,
+                   info->trap->name, window_class, window_name);
+            break;
+        case OUTPUT_JSON:
+            escaped_pname = drakvuf_escape_str(info->proc_data.name);
+            printf( "{"
+                    "\"Plugin\" : \"windowmon\","
+                    "\"TimeStamp\" :" "\"" FORMAT_TIMEVAL "\","
+                    "\"ProcessName\": %s,"
+                    "\"UserName\": \"%s\","
+                    "\"UserId\": %" PRIu64 ","
+                    "\"PID\" : %d,"
+                    "\"PPID\": %d,"
+                    "\"Method\" : \"%s\","
+                    "\"Class\" : %s,"
+                    "\"Name\" : %s,"
+                    "}\n",
+                    UNPACK_TIMEVAL(info->timestamp),
+                    escaped_pname,
+                    USERIDSTR(drakvuf), info->proc_data.userid,
+                    info->proc_data.pid, info->proc_data.ppid,
+                    info->trap->name,
+                    window_class, window_name);
+            g_free(escaped_pname);
+            break;
+        default:
+        case OUTPUT_DEFAULT:
+            printf("[WINDOWMON] TIME:" FORMAT_TIMEVAL " VCPU:%" PRIu32 " CR3:0x%" PRIx64 ",\"%s\" %s:%" PRIi64" CLASS:%s NAME:%s\n",
+                   UNPACK_TIMEVAL(info->timestamp), info->vcpu, info->regs->cr3, info->proc_data.name,
+                   USERIDSTR(drakvuf), info->proc_data.userid, window_class, window_name);
+            break;
+    }
+
+    vmi_free_unicode_str(class_us);
+    vmi_free_unicode_str(name_us);
+    g_free((gpointer)window_class);
+    g_free((gpointer)window_name);
+
+    return VMI_EVENT_RESPONSE_NONE;
+}
+
+static bool get_constant_rva(json_object* profile_json, const char* constant, addr_t* rva)
+{
+    if (!profile_json)
+    {
+        fprintf(stderr, "Rekall profile json is NULL!\n");
+        return false;
+    }
+
+    json_object* constants = NULL, *jsymbol = NULL;
+    if (!json_object_object_get_ex(profile_json, "$CONSTANTS", &constants))
+    {
+        PRINT_DEBUG("Rekall profile: no $CONSTANTS section found\n");
+        return false;
+    }
+
+    if (!json_object_object_get_ex(constants, constant, &jsymbol))
+    {
+        PRINT_DEBUG("Rekall profile: no '%s' found\n", constant);
+        return false;
+    }
+
+    *rva = json_object_get_int64(jsymbol);
+    return true;
+}
+
+struct SYSTEM_SERVICE_TABLE_32
```

Comment:
Please move these structure definitions into a `private.h`

Reply:
In fact, I don't see these structures actually being used for anything other then to calculate their size, so you might as well just define their size in a `#define`

Reply:
Done.